components/Document/api/fetchPostFromS3.ts

```diff
@@ -14,7 +15,29 @@ const fetchPostFromS3 = async ({ s3Url, cleanIntroEmptyContent = true }: Props):
         msg: `Error fetching post from S3`,
       });
     }
-    let _html = await response.text();
+    let _html = await response.text().then((text) =>
```

Comment:
This could work a little different than expected but the main motivation to use `await` is to circumvent `then` and avoid the javascript [Pyramid of Doom](https://blog.stackademic.com/callback-hell-callback-pyramids-in-javscript-d66e6ba27736).

In other words, you may be able to just do: `sanitizeHtml(text, {...` in subsequent line without being scoped to `then`